cookbooks/kinematically_driven_subduction_2d/subduction_plate_cooling.cc

```diff
@@ -0,0 +1,99 @@
+/*
+  Copyright (C) 2011 - 2023 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include "subduction_plate_cooling.h"
+#include <aspect/geometry_model/box.h>
+
+
+namespace aspect
+{
+  namespace InitialTemperature
+  {
+    template <int dim>
+    double
+    SubductionPlateCooling<dim>::
+    initial_temperature (const Point<dim> &position) const
+    {
+      // this initial condition only makes sense if the geometry is a
+      // Box. verify that it is indeed
+      AssertThrow(Plugins::plugin_type_matches<const GeometryModel::Box<dim>>(this->get_geometry_model()),
+                  ExcMessage("Subduction plate cooling initial temperature is not implemented "
+                             "for any other geometry model than a box. ") );
+
+      // Model domain is 3000 x 670 km.
```

Comment:
That would be easier for the user, but since this plugin is just for the cookbook reproducing the benchmark setup, and there already is a plate cooling option in the adiabatic initial temperature plugin, I'd prefer not to make these variables parsable from the input file. Would that be okay?